### PR TITLE
added some real world examples for patterns from stackoverflow

### PR DESCRIPTION
**Pull request by gansai**
http://stackoverflow.com/questions/1673841/examples-of-gof-design-patterns-in-javas-core-libraries

TODO: add hyperlink for each of the example added, so that it is browsable.

### CODE REVIEW

**Comment by gansai** [comment]
Hi @iluwatar,

I have added hyperlinks. 

Cheers,
Ganesh

File: README.md [hunk 1]
```diff
@@ -105,9 +108,14 @@ A programming idiom is a means of expressing a recurring construct in one or mor
 * the construction process must allow different representations for the object that's constructed
 
 **Real world examples:**
-* [java.lang.StringBuilder](http://docs.oracle.com/javase/8/docs/api/java/lang/StringBuilder.html)
+* [java.lang.StringBuilder#append() (unsynchronized)] (http://docs.oracle.com/javase/8/docs/api/java/lang/StringBuilder.html)
```
Comment (iluwatar) [inline README.md]: Please link only the method i.e. 

[java.lang.StringBuilder#append() (unsynchronized)] (http://docs.oracle.com/javase/8/docs/api/java/lang/StringBuilder.html)

change to:

[java.lang.StringBuilder#append()](http://docs.oracle.com/javase/8/docs/api/java/lang/StringBuilder.html) (unsynchronized)

File: README.md [hunk 2]
```diff
@@ -105,9 +108,14 @@ A programming idiom is a means of expressing a recurring construct in one or mor
...
+* [java.lang.StringBuffer#append() (synchronized)](http://docs.oracle.com/javase/7/docs/api/java/lang/StringBuffer.html)
```
Comment (iluwatar) [inline README.md]: Please link only the method

File: README.md [hunk 3]
```diff
@@ -105,9 +108,14 @@ A programming idiom is a means of expressing a recurring construct in one or mor
...
+* [java.nio.ByteBuffer#put() (also on CharBuffer, ShortBuffer, IntBuffer, LongBuffer, FloatBuffer and DoubleBuffer)](http://docs.oracle.com/javase/7/docs/api/java/nio/ByteBuffer.html)
```
Comment (iluwatar) [inline README.md]: Please link only the method

File: README.md [hunk 4]
```diff
@@ -105,9 +108,14 @@ A programming idiom is a means of expressing a recurring construct in one or mor
...
+* [javax.swing.GroupLayout.Group#addComponent()](https://docs.oracle.com/javase/6/docs/api/javax/swing/GroupLayout.Group.html)
+* [All implementations of java.lang.Appendable](http://docs.oracle.com/javase/7/docs/api/java/lang/Appendable.html)
```
Comment (iluwatar) [inline README.md]: Please link only the method

File: README.md [hunk 5]
```diff
@@ -120,6 +128,11 @@ A programming idiom is a means of expressing a recurring construct in one or mor
 
 **Real world examples:**
 * [java.util.Calendar#getInstance()](http://docs.oracle.com/javase/8/docs/api/java/util/Calendar.html#getInstance%28%29)
+* [java.util.ResourceBundle#getBundle()](https://docs.oracle.com/javase/8/docs/api/java/util/ResourceBundle.html#getBundle-java.lang.String-)
+* [java.text.NumberFormat#getInstance()](https://docs.oracle.com/javase/8/docs/api/java/text/NumberFormat.html#getInstance--)
+* [java.nio.charset.Charset#forName()](https://docs.oracle.com/javase/7/docs/api/java/nio/charset/Charset.html#forName(java.lang.String))
+* [java.net.URLStreamHandlerFactory#createURLStreamHandler(String) (Returns singleton object per protocol)](http://docs.oracle.com/javase/7/docs/api/java/net/URLStreamHandlerFactory.html#createURLStreamHandler(java.lang.String))
```
Comment (iluwatar) [inline README.md]: Please link only the method

File: README.md [hunk 6]
```diff
@@ -163,6 +177,9 @@ A programming idiom is a means of expressing a recurring construct in one or mor
 
 **Real world examples:**
 * [java.util.Arrays#asList()](http://docs.oracle.com/javase/8/docs/api/java/util/Arrays.html#asList%28T...%29)
+* [java.io.InputStreamReader(InputStream) (returns a Reader)](https://docs.oracle.com/javase/8/docs/api/java/io/InputStreamReader.html)
```
Comment (iluwatar) [inline README.md]: Please link only the method

File: README.md [hunk 7]
```diff
@@ -163,6 +177,9 @@ A programming idiom is a means of expressing a recurring construct in one or mor
...
+* [java.io.OutputStreamWriter(OutputStream) (returns a Writer)](https://docs.oracle.com/javase/8/docs/.../java/io/OutputStreamWriter.html)
```
Comment (iluwatar) [inline README.md]: Please link only the method

File: README.md [hunk 8]
```diff
@@ -163,6 +177,9 @@ A programming idiom is a means of expressing a recurring construct in one or mor
...
+* [javax.xml.bind.annotation.adapters.XmlAdapter#marshal() and #unmarshal()](https://docs.oracle.com/javase/8/docs/api/javax/xml/bind/annotation/adapters/XmlAdapter.html)
```
Comment (iluwatar) [inline README.md]: Please link only the method

File: README.md [hunk 9]
```diff
@@ -177,6 +194,9 @@ A programming idiom is a means of expressing a recurring construct in one or mor
 * you have a proliferation of classes. Such a class hierarchy indicates the need for splitting an object into two parts. Rumbaugh uses the term "nested generalizations" to refer to such class hierarchies
 * you want to share an implementation among multiple objects (perhaps using reference counting), and this fact should be hidden from the client. A simple example is Coplien's String class, in which multiple objects can share the same string representation.
 
+**Real world examples:**
+* [java.util.Collections#newSetFromMap() and singletonXXX() methods however comes close](https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#newSetFromMap-java.util.Map-)
```
Comment (iluwatar) [inline README.md]: Please link only the method

File: README.md [hunk 10]
```diff
@@ -189,6 +209,7 @@ A programming idiom is a means of expressing a recurring construct in one or mor
 **Real world examples:**
 * [java.awt.Container](http://docs.oracle.com/javase/8/docs/api/java/awt/Container.html) and [java.awt.Component](http://docs.oracle.com/javase/8/docs/api/java/awt/Component.html)
 * [Apache Wicket](https://github.com/apache/wicket) component tree, see [Component](https://github.com/apache/wicket/blob/91e154702ab1ff3481ef6cbb04c6044814b7e130/wicket-core/src/main/java/org/apache/wicket/Component.java) and [MarkupContainer](https://github.com/apache/wicket/blob/b60ec64d0b50a611a9549809c9ab216f0ffa3ae3/wicket-core/src/main/java/org/apache/wicket/MarkupContainer.java)
+* [javax.faces.component.UIComponent#getChildren() (practically all over JSF UI thus)](http://docs.oracle.com/javaee/6/api/javax/faces/component/UIComponent.html#getChildren())
```
Comment (iluwatar) [inline README.md]: Please link only the method

File: README.md [hunk 11]
```diff
@@ -200,6 +221,12 @@ A programming idiom is a means of expressing a recurring construct in one or mor
 * for responsibilities that can be withdrawn
 * when extension by subclassing is impractical. Sometimes a large number of independent extensions are possible and would produce an explosion of sublasses to support every combination. Or a class definition may be hidden or otherwise unavailable for subclassing
 
+**Real world examples:**
+* [All subclasses of java.io.InputStream, OutputStream, Reader and Writer have a constructor taking an instance of same type.](https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html)
```
Comment (iluwatar) [inline README.md]: Please link only the method

File: README.md [hunk 12]
```diff
@@ -200,6 +221,12 @@ A programming idiom is a means of expressing a recurring construct in one or mor
...
+* [java.util.Collections, the checkedXXX(), synchronizedXXX() and unmodifiableXXX() methods.](https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html)
```
Comment (iluwatar) [inline README.md]: Please link only the method

File: README.md [hunk 13]
```diff
@@ -200,6 +221,12 @@ A programming idiom is a means of expressing a recurring construct in one or mor
...
+* [javax.servlet.http.HttpServletRequestWrapper and HttpServletResponseWrapper](https://docs.oracle.com/javaee/6/api/javax/servlet/http/HttpServletRequestWrapper.html)
```
Comment (iluwatar) [inline README.md]: Please link only the method

File: README.md [hunk 14]
```diff
@@ -210,6 +237,11 @@ A programming idiom is a means of expressing a recurring construct in one or mor
 * there are many dependencies between clients and the implementation classes of an abstraction. Introduce a facade to decouple the subsystem from clients and other subsystems, thereby promoting subsystem independence and portability.
 * you want to layer your subsystems. Use a facade to define an entry point to each subsystem level. If subsystems are dependent, the you can simplify the dependencies between them by making them communicate with each other solely through their facades
 
+**Real world examples:**
+* [javax.faces.context.FacesContext, it internally uses among others the abstract/interface types LifeCycle, ViewHandler, NavigationHandler and many more without that the enduser has to worry about it (which are however overrideable by injection](http://docs.oracle.com/javaee/6/api/javax/faces/context/FacesContext.html)
```
Comment (iluwatar) [inline README.md]: Please link only the method

File: README.md [hunk 15]
```diff
@@ -210,6 +237,11 @@ A programming idiom is a means of expressing a recurring construct in one or mor
...
+* [javax.faces.context.ExternalContext, which internally uses ServletContext, HttpSession, HttpServletRequest, HttpServletResponse, etc.](http://docs.oracle.com/javaee/6/api/javax/faces/context/ExternalContext.html)
```
Comment (iluwatar) [inline README.md]: Please link only the method

File: README.md [hunk 16]
```diff
@@ -246,6 +278,7 @@ A programming idiom is a means of expressing a recurring construct in one or mor
 
 **Real world examples:**
 * [java.lang.reflect.Proxy](http://docs.oracle.com/javase/8/docs/api/java/lang/reflect/Proxy.html)
+* [java.rmi.*, the whole API actually.](http://docs.oracle.com/javase/8/docs/api/java/rmi/package-summary.html)
```
Comment (iluwatar) [inline README.md]: Please link only the method

File: README.md [hunk 17]
```diff
@@ -304,6 +339,14 @@ A programming idiom is a means of expressing a recurring construct in one or mor
 * the grammar is simple. For complex grammars, the class hierarchy for the grammar becomes large and unmanageable. Tools such as parser generators are a better alternative in such cases. They can interpret expressions without building abstract syntax trees, which can save space and possibly time
 * efficiency is not a critical concern. The most efficient interpreters are usually not implemented by interpreting parse trees directly but by first translating them into another form. For example, regular expressions are often transformed into state machines. But even then, the translator can be implemented by the Interpreter pattern, so the pattern is still applicable
 
+**Real world examples:**
+* [java.util.Pattern](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html)
+* [java.text.Normalizer](http://docs.oracle.com/javase/8/docs/api/java/text/Normalizer.html)
+* [All subclasses of java.text.Format](https://docs.oracle.com/javase/8/docs/api/java/text/Format.html)
```
Comment (iluwatar) [inline README.md]: Please link only the method

File: README.md [hunk 18]
```diff
@@ -304,6 +339,14 @@ A programming idiom is a means of expressing a recurring construct in one or mor
...
+* [All subclasses of javax.el.ELResolver](http://docs.oracle.com/javaee/6/api/javax/el/ELResolver.html)
```
Comment (iluwatar) [inline README.md]: Please link only the method

File: README.md [hunk 19]
```diff
@@ -316,6 +359,7 @@ A programming idiom is a means of expressing a recurring construct in one or mor
 
 **Real world examples:**
 * [java.util.Iterator](http://docs.oracle.com/javase/8/docs/api/java/util/Iterator.html)
+* [All implementations of java.util.Enumeration](https://docs.oracle.com/javase/8/docs/api/java/util/Enumeration.html)
```
Comment (iluwatar) [inline README.md]: Please link only the method

File: README.md [hunk 20]
```diff
@@ -327,6 +371,14 @@ A programming idiom is a means of expressing a recurring construct in one or mor
 * reusing an object is difficult because it refers to and communicates with many other objects
 * a behavior that's distributed between several classes should be customizable without a lot of subclassing
 
+**Real world examples:**
+* [java.util.Timer (all scheduleXXX() methods)](https://docs.oracle.com/javase/8/docs/api/java/util/Timer.html)
```
Comment (iluwatar) [inline README.md]: Please link only the method

File: README.md [hunk 21]
```diff
@@ -327,6 +371,14 @@ A programming idiom is a means of expressing a recurring construct in one or mor
...
+* [java.util.concurrent.Executor#execute()](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/Executor.html)
+* [java.util.concurrent.ExecutorService (the invokeXXX() and submit() methods)](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ExecutorService.html)
```
Comment (iluwatar) [inline README.md]: Please link only the method

File: README.md [hunk 22]
```diff
@@ -327,6 +371,14 @@ A programming idiom is a means of expressing a recurring construct in one or mor
...
+* [java.util.concurrent.ScheduledExecutorService (all scheduleXXX() methods)](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ScheduledExecutorService.html)
```
Comment (iluwatar) [inline README.md]: Please link only the method

File: README.md [hunk 23]
```diff
@@ -337,7 +389,10 @@ A programming idiom is a means of expressing a recurring construct in one or mor
 * a direct interface to obtaining the state would expose implementation details and break the object's encapsulation
 
 **Real world examples:**
-* [java.util.Date](http://docs.oracle.com/javase/8/docs/api/java/util/Date.html)
+* [java.util.Date](http://docs.oracle.com/javase/8/docs/api/java/util/Date.html) (the setter methods do that, Date is internally represented by a long value)
+* [All implementations of java.io.Serializable](https://docs.oracle.com/javase/8/docs/api/java/io/Serializable.html)
```
Comment (iluwatar) [inline README.md]: Please link only the method

File: README.md [hunk 24]
```diff
@@ -337,7 +389,10 @@ A programming idiom is a means of expressing a recurring construct in one or mor
...
+* [All implementations of javax.faces.component.StateHolder](http://docs.oracle.com/javaee/6/api/javax/faces/component/StateHolder.html)
```
Comment (iluwatar) [inline README.md]: Please link only the method

File: README.md [hunk 25]
```diff
@@ -356,6 +411,11 @@ A programming idiom is a means of expressing a recurring construct in one or mor
 
 **Real world examples:**
 * [java.util.Observer](http://docs.oracle.com/javase/8/docs/api/java/util/Observer.html)
+* [All implementations of java.util.EventListener (practically all over Swing thus)](https://docs.oracle.com/javase/6/docs/api/java/util/EventListener.html)
```
Comment (iluwatar) [inline README.md]: Please link only the method

File: README.md [hunk 26]
```diff
@@ -366,6 +426,9 @@ A programming idiom is a means of expressing a recurring construct in one or mor
 * an object's behavior depends on its state, and it must change its behavior at run-time depending on that state
 * operations have large, multipart conditional statements that depend on the object's state. This state is usually represented by one or more enumerated constants. Often, several operations will contain this same conditional structure. The State pattern puts each branch of the conditional in a separate class. This lets you treat the object's state as an object in its own right that can vary independently from other objects.
 
+**Real world examples:**
+* [javax.faces.lifecycle.LifeCycle#execute() (controlled by FacesServlet, the behaviour is dependent on current phase (state) of JSF lifecycle)](http://docs.oracle.com/cd/E17802_01/j2ee/j2ee/javaserverfaces/1.2/docs/api/javax/faces/lifecycle/Lifecycle.html)
```
Comment (iluwatar) [inline README.md]: Please link only the method

File: README.md [hunk 27]
```diff
@@ -377,6 +440,12 @@ A programming idiom is a means of expressing a recurring construct in one or mor
 * an algorithm uses data that clients shouldn't know about. Use the Strategy pattern to avoid exposing complex, algorithm-specific data structures
 * a class defines many behaviors, and these appear as multiple conditional statements in its operations. Instead of many conditionals, move related conditional branches into their own Strategy class
 
+**Real world examples:**
+* [java.util.Comparator#compare(), executed by among others Collections#sort()](https://docs.oracle.com/javase/8/docs/api/java/util/Comparator.html)
```
Comment (iluwatar) [inline README.md]: Please link only the method

File: README.md [hunk 28]
```diff
@@ -377,6 +440,12 @@ A programming idiom is a means of expressing a recurring construct in one or mor
...
+* [javax.servlet.http.HttpServlet, the service() and all doXXX() methods take HttpServletRequest and HttpServletResponse and the implementor has to process them (and not to get hold of them as instance variables!)](https://tomcat.apache.org/tomcat-5.5-doc/servletapi/javax/servlet/http/HttpServlet.html)
```
Comment (iluwatar) [inline README.md]: Please link only the method

File: README.md [hunk 29]
```diff
@@ -387,6 +456,12 @@ A programming idiom is a means of expressing a recurring construct in one or mor
 * when common behavior among subclasses should be factored and localized in a common class to avoid code duplication. This is good example of "refactoring to generalize" as described by Opdyke and Johnson. You first identify the differences in the existing code and then separate the differences into new operations. Finally, you replace the differing code with a template method that calls one of these new operations
 * to control subclasses extensions. You can define a template method that calls "hook" operations at specific points, thereby permitting extensions only at those points
 
+**Real world examples:**
+* [All non-abstract methods of java.io.InputStream, java.io.OutputStream, java.io.Reader and java.io.Writer.](http://docs.oracle.com/javase/7/docs/api/java/io/InputStream.html)
```
Comment (iluwatar) [inline README.md]: Please link only the method

File: README.md [hunk 30]
```diff
@@ -387,6 +456,12 @@ A programming idiom is a means of expressing a recurring construct in one or mor
...
+* [All non-abstract methods of java.util.AbstractList, java.util.AbstractSet and java.util.AbstractMap.](https://docs.oracle.com/javase/7/docs/api/java/util/AbstractList.html)
```
Comment (iluwatar) [inline README.md]: Please link only the method

File: README.md [hunk 31]
```diff
@@ -387,6 +456,12 @@ A programming idiom is a means of expressing a recurring construct in one or mor
...
+* [javax.servlet.http.HttpServlet, all the doXXX() methods by default sends a HTTP 405 "Method Not Allowed" error to the response. You're free to implement none or any of them.](https://tomcat.apache.org/tomcat-5.5-doc/servletapi/javax/servlet/http/HttpServlet.html)
```
Comment (iluwatar) [inline README.md]: Please link only the method

File: README.md [hunk 32]
```diff
@@ -508,7 +587,7 @@ A programming idiom is a means of expressing a recurring construct in one or mor
 * eager loading is expensive or the object to be loaded might not be need at all
 
 **Real world examples:**
-* JPA annotations @OneToOne, @OneToMany, @ManyToOne, @ManyToMany and fetch = FetchType.LAZY
+* [JPA annotations @OneToOne, @OneToMany, @ManyToOne, @ManyToMany and fetch = FetchType.LAZY](http://www.objectdb.com/api/java/jpa/annotations)
```
Comment (iluwatar) [inline README.md]: Please link only the method

**Comment by iluwatar** [comment]
@gansai In general, I'm not comfortable with linking style such as this:

[java.util.Collections#newSetFromMap() and singletonXXX() methods however comes close](https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#newSetFromMap-java.util.Map-)

The correct style looks like this:

[java.util.Collections#newSetFromMap()](https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#newSetFromMap-java.util.Map-) and singletonXXX() methods however comes close

**Comment by gansai** [comment]
Hi @iluwatar ,

Could you please check my changes?

Thanks,
gansai

File: README.md [hunk 33]
```diff
@@ -200,6 +227,17 @@ A programming idiom is a means of expressing a recurring construct in one or mor
 * for responsibilities that can be withdrawn
 * when extension by subclassing is impractical. Sometimes a large number of independent extensions are possible and would produce an explosion of sublasses to support every combination. Or a class definition may be hidden or otherwise unavailable for subclassing
 
+**Real world examples:**
+* [All subclasses of java.io.InputStream](https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html)
```
Comment (iluwatar) [inline README.md]: Link style

File: README.md [hunk 34]
```diff
@@ -200,6 +227,17 @@ A programming idiom is a means of expressing a recurring construct in one or mor
...
+* [All subclasses of OutputStream](https://docs.oracle.com/javase/8/docs/api/java/io/OutputStream.html)
```
Comment (iluwatar) [inline README.md]: Link style

File: README.md [hunk 35]
```diff
@@ -200,6 +227,17 @@ A programming idiom is a means of expressing a recurring construct in one or mor
...
+* [All subclasses of Reader](https://docs.oracle.com/javase/8/docs/api/java/io/Reader.html)
```
Comment (iluwatar) [inline README.md]: Link style

File: README.md [hunk 36]
```diff
@@ -200,6 +227,17 @@ A programming idiom is a means of expressing a recurring construct in one or mor
...
+* [All subclasses of Writer](https://docs.oracle.com/javase/8/docs/api/java/io/Writer.html)
```
Comment (iluwatar) [inline README.md]: Link style

File: README.md [hunk 37]
```diff
@@ -294,6 +339,7 @@ A programming idiom is a means of expressing a recurring construct in one or mor
 
 **Real world examples:**
 * [java.lang.Runnable](http://docs.oracle.com/javase/8/docs/api/java/lang/Runnable.html)
+* [All implementations of javax.swing.Action](https://docs.oracle.com/javase/8/docs/api/javax/swing/Action.html)
```
Comment (iluwatar) [inline README.md]: Link style

File: README.md [hunk 38]
```diff
@@ -356,6 +423,11 @@ A programming idiom is a means of expressing a recurring construct in one or mor
 
 **Real world examples:**
 * [java.util.Observer](http://docs.oracle.com/javase/8/docs/api/java/util/Observer.html)
+* [All implementations of java.util.EventListener](https://docs.oracle.com/javase/6/docs/api/java/util/EventListener.html)
```
Comment (iluwatar) [inline README.md]: Link style

File: README.md [hunk 39]
```diff
@@ -387,6 +469,17 @@ A programming idiom is a means of expressing a recurring construct in one or mor
 * when common behavior among subclasses should be factored and localized in a common class to avoid code duplication. This is good example of "refactoring to generalize" as described by Opdyke and Johnson. You first identify the differences in the existing code and then separate the differences into new operations. Finally, you replace the differing code with a template method that calls one of these new operations
 * to control subclasses extensions. You can define a template method that calls "hook" operations at specific points, thereby permitting extensions only at those points
 
+**Real world examples:**
+* [All non-abstract methods of java.io.InputStream](http://docs.oracle.com/javase/7/docs/api/java/io/InputStream.html)
```
Comment (iluwatar) [inline README.md]: Link style

File: README.md [hunk 40]
```diff
@@ -387,6 +469,17 @@ A programming idiom is a means of expressing a recurring construct in one or mor
...
+* [All non-abstract methods of java.io.OutputStream](http://docs.oracle.com/javase/7/docs/api/java/io/OutputStream.html)
```
Comment (iluwatar) [inline README.md]: Link style

File: README.md [hunk 41]
```diff
@@ -387,6 +469,17 @@ A programming idiom is a means of expressing a recurring construct in one or mor
...
+* [All non-abstract methods of java.io.Reader](http://docs.oracle.com/javase/7/docs/api/java/io/Reader.html)
```
Comment (iluwatar) [inline README.md]: Link style

File: README.md [hunk 42]
```diff
@@ -387,6 +469,17 @@ A programming idiom is a means of expressing a recurring construct in one or mor
...
+* [All non-abstract methods of java.io.Writer](http://docs.oracle.com/javase/7/docs/api/java/io//Writer.html)
```
Comment (iluwatar) [inline README.md]: Link style

File: README.md [hunk 43]
```diff
@@ -387,6 +469,17 @@ A programming idiom is a means of expressing a recurring construct in one or mor
...
+* [All non-abstract methods of java.util.AbstractList](https://docs.oracle.com/javase/7/docs/api/java/util/AbstractList.html)
```
Comment (iluwatar) [inline README.md]: Link style

File: README.md [hunk 44]
```diff
@@ -387,6 +469,17 @@ A programming idiom is a means of expressing a recurring construct in one or mor
...
+* [All non-abstract methods of java.util.AbstractSet](https://docs.oracle.com/javase/7/docs/api/java/util/AbstractSet.html)
```
Comment (iluwatar) [inline README.md]: Link style

File: README.md [hunk 45]
```diff
@@ -387,6 +469,17 @@ A programming idiom is a means of expressing a recurring construct in one or mor
...
 * to control subclasses extensions. You can define a template method that calls "hook" operations at specific points, thereby permitting extensions only at those points
 
+**Real world examples:**
+* [All non-abstract methods of java.io.InputStream](http://docs.oracle.com/javase/7/docs/api/java/io/InputStream.html)
+* [All non-abstract methods of java.io.OutputStream](http://docs.oracle.com/javase/7/docs/api/java/io/OutputStream.html)
+* [All non-abstract methods of java.io.Reader](http://docs.oracle.com/javase/7/docs/api/java/io/Reader.html)
+* [All non-abstract methods of java.io.Writer](http://docs.oracle.com/javase/7/docs/api/java/io//Writer.html)
+* [All non-abstract methods of java.util.AbstractList](https://docs.oracle.com/javase/7/docs/api/java/util/AbstractList.html)
+* [All non-abstract methods of java.util.AbstractSet](https://docs.oracle.com/javase/7/docs/api/java/util/AbstractSet.html) 
+* [All non-abstract methods of java.util.AbstractMap.](https://docs.oracle.com/javase/7/docs/api/java/util/AbstractMap.html)
```
Comment (iluwatar) [inline README.md]: Link style

**Comment by iluwatar** [comment]
@gansai I added my comments. Please double check all the links.

**Pull request closed without merging**